stdlib/experimental/prometheus/scrape.go

```diff
@@ -0,0 +1,495 @@
+
+			// Metric Type: Histogram
+			case dto.MetricType_HISTOGRAM:
+				makeMetrics := p.makeBuckets(metr, tags, field, family.GetType())
+				p.metrics = append(p.metrics, makeMetrics...)
+
+			// Metric Type: Gague, Counter, Untyped
+			default:
+				typeValue := getNameAndValue(metr, field)
+
+				if len(typeValue) > 0 {
+					var t time.Time
+					if metr.TimestampMs != nil && *metr.TimestampMs > 0 {
+						t = time.Unix(0, *metr.TimestampMs*msToNanoseconds)
+					} else {
+						t = p.now
+					}
+					met := Metric{
+						Timestamp: t,
+						Tags:      tags,
+						TypeVal:   typeValue,
+						Field:     field,
+						Type:      family.GetType().String(),
+					}
+					p.metrics = append(p.metrics, met)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// This implementation of Fetch will iterate over p.metrics
+func (p *PrometheusIterator) Fetch(ctx context.Context) (bool, error) {
+	// Iterate over all Metrics
+	if p.i < len(p.metrics) {
+		return true, nil
+	}
+
+	// No more metrics to return
+	return false, nil
+}
+
+// This implementation of Decode will create flux Tables for a give Metric. It retrieves one Metric
+// from p.metrics and places it into a flux.Table
+func (p *PrometheusIterator) Decode(ctx context.Context) (table flux.Table, err error) {
+	met := p.metrics[p.i]
+
+	// Unpacking TypeVal map
+	var val interface{}
+	for _, v := range met.TypeVal {
+		val = v
+	}
+
+	groupKey := execute.NewGroupKeyBuilder(nil)
+	groupKey.AddKeyValue("_measurement", values.New("prometheus"))
+	groupKey.AddKeyValue("_field", values.New(met.Field))
+
+	// Add all tag names to Group Key
+	gkInt := 2
+	for name, val := range met.Tags {
+		gkInt++
+		if groupKey.Len() < gkInt {
+			groupKey.AddKeyValue(name, values.New(val))
+		}
+	}
+
+	gk, err := groupKey.Build()
+	if err != nil {
+		return nil, err
+	}
+
+	builder := execute.NewColListTableBuilder(gk, p.administration.Allocator())
+
+	builder.AddCol(flux.ColMeta{
+		Label: "_time",
+		Type:  flux.TTime,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "_value",
+		Type:  flux.TFloat,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "_measurement", // data source
+		Type:  flux.TString,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "_field", // prometheus metric name
+		Type:  flux.TString,
+	})
+	builder.AddCol(flux.ColMeta{
+		Label: "url",
+		Type:  flux.TString,
+	})
+
+	// Add all tags to Col list
+	for name := range met.Tags {
+		if execute.ColIdx(name, builder.Cols()) == -1 {
+			builder.AddCol(flux.ColMeta{
+				Label: name,
+				Type:  flux.TString,
+			})
+		}
+	}
+
+	builder.AppendTime(0, values.ConvertTime(met.Timestamp))
+	builder.AppendValue(1, values.New(val))
+	builder.AppendValue(2, values.New("prometheus"))
+	builder.AppendValue(3, values.New(met.Field))
+	builder.AppendValue(4, values.New(p.url))
+
+	// Add tag values
+	for name, tagVal := range met.Tags {
+		builder.AppendValue(execute.ColIdx(name, builder.Cols()), values.New(tagVal))
+	}
+
+	// Grab the next metric in list
+	p.i++
```

Comment:
I did try to keep this in fetch. It is tricky because of the way that fetch and decode interact. There isn't really a good place in fetch for this line without causing panics.

Reply:
Also, I did show the latest version to nathaniel and goller before submitting this PR. Goller seemed okay with the output.